ykrt/src/compile/jitc_yk/codegen/x64/deopt.rs

```diff
@@ -42,6 +42,20 @@ const RECOVER_REG: [usize; 31] = [
 /// from zero). This is used to allocate arrays whose indices need to be the DWARF register number.
 const REGISTER_NUM: usize = RECOVER_REG.len() + 2;
 
+#[unsafe(no_mangle)]
+pub(crate) extern "C" fn __yk_ret_from_trace(ctrid: u64) {
+    let ctr = MTThread::with_borrow(|mtt| mtt.compiled_trace(TraceId::from_u64(ctrid)))
+        .as_any()
+        .downcast::<X64CompiledTrace>()
+        .unwrap();
+    let mt = Arc::clone(&ctr.mt);
+    mt.deopt();
+    mt.stats
+        .timing_state(crate::log::stats::TimingState::OutsideYk);
+    mt.log
```

Comment:
This reminds me: we really need to fix logging, because at the moment we often generate trsing with `format` then `log` says "no logging needed" and throws them away. That must cost us.